### PR TITLE
Niu/prompt dialog

### PR DESCRIPTION
**Pull request by xiaoniuniu89**
# ✨ Codu Pull Request 💻

Codu Logo

👉 _**Please remove the below and replace with your own values, leaving the headers where they are.**_ 👈

## Pull Request details:
This pr updates prompt dialog for unsaved changes to work with next 14 and the app router. It also includes some updated abstractions to make it more reusable eg: delete article. 

## Any Breaking changes:
none - not implemented in alpha/new editor

## Associated Screenshots:
    
_( Welcome file extensions include gifs/png screenshots of your feature in action )_

- IF YOU HAVE ANY SCREENSHOTS, INCLUDE THEM HERE.
- IF YOU HAVE NO SCREENSHOTS, ENTER 'None'

### CODE REVIEW

File: components/Nav/Nav.tsx [hunk 1]
```diff
@@ -111,8 +111,8 @@ const Nav = ({ session }: { session: Session | null }) => {
                   {session && (
                     <>
                       <Link
-                        title="Notifications"
-                        href="/notifications"
+                        // title="Notifications"
```
Comment (NiallJoeMaher) [inline components/Nav/Nav.tsx]: Is this meant to be left in? 🤔

File: components/Nav/Nav.tsx [hunk 2]
```diff
@@ -180,8 +180,8 @@ const Nav = ({ session }: { session: Session | null }) => {
                 <ThemeToggle />
                 {session && (
                   <Link
-                    title="Notifications"
-                    href="/notifications"
+                    // title="Notifications"
```
Comment (NiallJoeMaher) [inline components/Nav/Nav.tsx]: Same?

**Review by NiallJoeMaher — COMMENTED** [review]
Looks good! Great work. Just a couple of comments to check.

**Review by NiallJoeMaher — APPROVED** [review]
🌮 Thanks! LGTM

**Comment by NiallJoeMaher** [comment]
Build error! I'll take a look in the morning and see if it's something obvious.

**Comment by xiaoniuniu89** [comment]
sure, can send me results from build log and I can have a look during the week,

**Comment by xiaoniuniu89** [comment]
Had some unresolved changes i didnt fix in alpha editor, should pass now.

**Comment by xiaoniuniu89** [comment]
@NiallJoeMaher I have passing build locally now - lint was failing, we can update pre commit hook for that

**Pull request merged**